examples/dump-ast.rs

```diff
@@ -0,0 +1,28 @@
+#![cfg(all(feature = "full", feature = "extra-traits"))]
```

Comment:
I think this may be best as a `required-features` entry in the manifest, otherwise I think this fails to compile unless the features are set?